src/panels/lovelace/cards/hui-weather-forecast-card.ts

```diff
@@ -0,0 +1,455 @@
+import {
+  html,
+  LitElement,
+  PropertyValues,
+  TemplateResult,
+  css,
+  CSSResult,
+  property,
+  customElement,
+} from "lit-element";
+
+import "../../../components/ha-card";
+import "../components/hui-warning";
+
+import isValidEntityId from "../../../common/entity/valid_entity_id";
+import computeStateName from "../../../common/entity/compute_state_name";
+
+import { HomeAssistant } from "../../../types";
+import { hasConfigOrEntityChanged } from "../common/has-changed";
+import { LovelaceCard, LovelaceCardEditor } from "../types";
+import { WeatherForecastCardConfig } from "./types";
+import { computeRTL } from "../../../common/util/compute_rtl";
+import { fireEvent } from "../../../common/dom/fire_event";
+
+const cardinalDirections = [
+  "N",
+  "NNE",
+  "NE",
+  "ENE",
+  "E",
+  "ESE",
+  "SE",
+  "SSE",
+  "S",
+  "SSW",
+  "SW",
+  "WSW",
+  "W",
+  "WNW",
+  "NW",
+  "NNW",
+  "N",
+];
+
+const weatherIcons = {
+  "clear-night": "hass:weather-night",
+  cloudy: "hass:weather-cloudy",
+  exceptional: "hass:alert-circle-outline",
+  fog: "hass:weather-fog",
+  hail: "hass:weather-hail",
+  lightning: "hass:weather-lightning",
+  "lightning-rainy": "hass:weather-lightning-rainy",
+  partlycloudy: "hass:weather-partly-cloudy",
+  pouring: "hass:weather-pouring",
+  rainy: "hass:weather-rainy",
+  snowy: "hass:weather-snowy",
+  "snowy-rainy": "hass:weather-snowy-rainy",
+  sunny: "hass:weather-sunny",
+  windy: "hass:weather-windy",
+  "windy-variant": "hass:weather-windy-variant",
+};
+
+@customElement("hui-weather-forecast-card")
+class HuiWeatherForecastCard extends LitElement implements LovelaceCard {
+  public static async getConfigElement(): Promise<LovelaceCardEditor> {
+    await import(/* webpackChunkName: "hui-weather-forecast-card-editor" */ "../editor/config-elements/hui-weather-forecast-card-editor");
+    return document.createElement("hui-weather-forecast-card-editor");
+  }
+  public static getStubConfig(): object {
+    return {};
```

Comment:
Add entity here, so we know that the card takes an entity
```suggestion
    return { entity: "" };
```